app/lib/kafka.server.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Kafka } from 'gcn-kafka'
+
+import { getEnvOrDie, hostname } from './env.server'
+import type { Circular } from '~/routes/circulars/circulars.lib'
+
+const client_id = getEnvOrDie('KAFKA_CLIENT_ID')
+const client_secret = getEnvOrDie('KAFKA_CLIENT_SECRET')
+const domain = hostname as
+  | 'gcn.nasa.gov'
+  | 'test.gcn.nasa.gov'
+  | 'dev.gcn.nasa.gov'
+
+const producer = new Kafka({
+  client_id,
+  client_secret,
+  domain,
+}).producer()
+
+export async function sendKafka(circular: Circular, topic: string) {
+  delete circular.sub
```

Comment:
Does modifying the Circular in place like that possibly change its value in other functions that might require `sub` to be present? If so, you could remove the `sub` field non-destructively by doing something like:

```export async function sendKafka({sub, ...circular}: Circular, topic: string)```

Also, I notice that this function is in the directory app/lib, so it shouldn't have any Circular-specific code. Move Circular-specific code into table-streams/circulars. You might have a general function to send _any_ Kafka record here and a function that calls it in table-streams/circulars.